x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/job/util/PageParams.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.ml.job.util;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Paging parameters for GET requests
+ */
+public class PageParams implements ToXContentObject {
+
+    public static final ParseField PAGE = new ParseField("page");
+    public static final ParseField FROM = new ParseField("from");
+    public static final ParseField SIZE = new ParseField("size");
+
+    public static final ConstructingObjectParser<PageParams, Void> PARSER = new ConstructingObjectParser<>(PAGE.getPreferredName(),
+            a -> new PageParams((Integer) a[0], (Integer) a[1]));
+
+    static {
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), FROM);
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), SIZE);
+    }
+
+    private final Integer from;
+    private final Integer size;
+
+    /**
+     * Constructs paging parameters
+     * @param from skips the specified number of buckets. When {@code null} the default value will be used.
```

Comment:
from skips the specified number of **buckets**

This class will be reused won't it so not necessarily buckets